integration-tests/test_file_workflow.py

```diff
@@ -114,6 +118,34 @@ def test_file_workflow_with_an_archive_without_canonical_facts(
     assert "Error: failed to find host with matching machine-id" in check_results.stdout
 
 
+def test_file_workflow_archive_update_host_info(insights_client, external_inventory):
+    """Verify updating files in an Insights Archive updates host
+    information on Inventory.
+    test_steps:
+            1. Register Insights-client and upload data
+            2. Update hostname and recollect the data
+            3. Upload the new archive
+            4. Retrieve the host data from Inventory and check display name
+    expected_results:
+            1. System is registered and archive successfully uploaded
+            2. display_name returned in step 5 matches updated hostname
+    """
+    insights_client.register()
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+    current_hostname = subprocess.check_output("hostname", shell=True).decode().strip()
+    try:
+        # set a new hostname
+        new_hostname = set_hostname()
+        # collect data and upload again
+        insights_client.run()
+        sleep(30)  # small wait for data to get reflected in inventory
```

Comment:
I'm gonna assume there's no better way to do this.

Reply:
unfortunately yes there is no other way at this point then being hopeful(sleep).